File: lw5/BatcherAlgorithm/CalculatorAgregator.cpp
#include "stdafx.h"
#include "CalculatorAgregator.h"
#include <ctime>

CCalculatorAggregator::CCalculatorAggregator(IBatcherCalculator & calc, const std::string & filename)
    : m_calc(calc)
{
    ReadVectorFromFile(filename);
}

void CCalculatorAggregator::Start()
{  
    SortArray(false);
    SortArray();
}

void CCalculatorAggregator::ReadVectorFromFile(const std::string & fileName)
{
    std::ifstream file(fileName);
    int elem;
    while (!file.eof())
    {
        file >> elem;
        m_fileVec.push_back(elem);
    }
}

void CCalculatorAggregator::CoutArray(const std::string & message, const std::vector<int>& vec)
{
    std::cout << message << std::endl;
}

void CCalculatorAggregator::SortArray(bool parallel)
{
    m_calc.Reset(m_fileVec);

    clock_t begin = clock();
    m_calc.Sort(parallel);
    clock_t end = clock();
    double elapsed_secs = double(end - begin) / CLOCKS_PER_SEC;

    CoutArray("Elapsed time: " + std::to_string(elapsed_secs) + ". Sorted", m_calc.GetResult());
}

File: lw7/2/Anikin_Oleg/ArraySort/ArraySort/CalculatorAgregator.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;

namespace ArraySort
{
    class CalculatorAgregator
    {
        private ISortCalculator m_calc;
        private List<int> m_fileVec;

        public CalculatorAgregator(ref ISortCalculator calc, string filename)
        {
            m_calc = calc;
            m_fileVec = new List<int>();
            ReadListFromFile(filename);
        }

        public void Start()
        {
            SortArray(true);
            SortArray(false);
        }

        private void ReadListFromFile(string fileName)
        {
            using (TextReader reader = File.OpenText(fileName))
            {
                string[] text = reader.ReadToEnd().Split(' ');
                for(int i = 0; i < text.Length; i++)
                {
                    m_fileVec.Add(Convert.ToInt32(text[i]));
                }
            }
        }

        private void SortArray(bool parallel)
        {
            m_calc.Reset(new List<int>(m_fileVec));
            m_calc.Sort(parallel);
        }
    }
}
